### PR TITLE
feature: Add eslint ts stylelint code check

### PR DESCRIPTION
**Pull request by songpengyuan**
### Ⅰ. Describe what this PR did
- Project Add LICENSE File
- Add code style check in front end (according to ice specification)
- Fix/ignore historical code style style

### Ⅱ. Does this pull request fix one issue?

#65 

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
Manual test

### Ⅳ. Describe how to verify it
Manual

### Ⅴ. Special notes for reviews
- New code check npm command
    - `npm run lint`
    - `npm run eslint`
    - `npm run eslint:fix`
    - `npm run stylelint`

### CODE REVIEW

**Review by CH3CHO — COMMENTED** [review]
Could you add a lint step to the github action?

By the way, could you give an example of lint failed? I tried to mess up some code lines but it still returned a pass.

image

File: frontend/.eslintignore [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+# node 覆盖率文件
+coverage/
+
+# 忽略文件
+**/*-min.js
+**/*.min.js
+
+
+
```
[the remark this answers is not in the document]
Comment (songpengyuan) [reply frontend/.eslintignore]: New commit: Add .ts file format eslint check

test Demo:
```js
const testEslint = (a, b) => {
  console.log(a); // Unexpected console statement         
  const c = 100; 
  c = 100; // 'c' is constant    
  return; // Unnecessary return statement   
};

testEslint; // Expected an assignment or function call and instead saw an expression
```

**Review by CH3CHO — COMMENTED** [review]
Looks good now. Could you add a lint step to the github action?

**Comment by songpengyuan** [comment]
feat: suppress errors in git hooks

image

**Comment by CH3CHO** [comment]
> feat: suppress errors in git hooks
> 
> image

Actually, I wasn't talking about Git hooks, but [GitHub action](https://github.com/features/actions).

You can talk a look at this: https://github.com/higress-group/higress-console/blob/main/.github/workflows/build-and-test.yaml

**Review by CH3CHO — CHANGES_REQUESTED** [review]
Actually, I wasn't talking about Git hooks, but [GitHub action](https://github.com/features/actions).

**Comment by songpengyuan** [comment]
> 实际上，我不是在谈论 Git hooks，而是[GitHub action](https://github.com/features/actions)。

Done

image

File: .github/workflows/frontend-code-checker.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  pull_request:
+    branches: ["*"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
```
Comment (CH3CHO) [inline .github/workflows/frontend-code-checker.yaml:15]: 我们现在打包用的是16，所以这里可以考虑只跑16节约时间。
另外这里跑不同的版本会产生不同的结果吗？

File: .github/workflows/frontend-code-checker.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: cd frontend && install
+        run: |
+          cd frontend && npm install
+      - name: check
```
Comment (CH3CHO) [inline .github/workflows/frontend-code-checker.yaml:28]: 为什么这个命名就不 cd frontend 了。。。

**Review by CH3CHO — CHANGES_REQUESTED** [review]

File: .github/workflows/frontend-code-checker.yaml (hunk 3 above)
Comment (songpengyuan) [reply .github/workflows/frontend-code-checker.yaml:28]: 我来修改下名字

cd frontend && npm run lint

**Review by CH3CHO — APPROVED** [review]
LGTM

**Pull request merged**